File: sankofa_app/lib/services/api_exception.dart
class ApiException implements Exception {
  ApiException(this.message, {this.statusCode, this.details});

  final String message;
  final int? statusCode;
  final Map<String, dynamic>? details;

  @override
  String toString() => 'ApiException($message, status: $statusCode, details: $details)';
}

File: sankofa_frontend/src/lib/apiException.ts
/**
 * API Exception class
 * Mirrors the mobile app's ApiException
 */

export class ApiException extends Error {
  statusCode?: number;
  details?: Record<string, unknown>;

  constructor(message: string, statusCode?: number, details?: Record<string, unknown>) {
    super(message);
    this.name = 'ApiException';
    this.statusCode = statusCode;
    this.details = details;
    Object.setPrototypeOf(this, ApiException.prototype);
  }
}
